### ISSUE TITLE
ceiling might be reached soon-ish ;)

### ISSUE THREAD

**Issue by iham** [plone/Products.Archetypes#133]
greets from @gogobd and me:

the calendar widget will store dates beyond 2020/12/31, but will not show it, when editing again.

reason: this - "silly" ceiling:
https://github.com/plone/Products.Archetypes/blob/master/Products/Archetypes/browser/datecomponents.py#L9

**Comment by iham** [comment]
@mauritsvanrees i know ... its archetypes, its old, ... but still in use, i guess.
can u release that?

**Comment by mauritsvanrees** [comment]
Sure. I have added it to the `checkouts.cfg` of coredev branch 5.2. When the tests pass, I wil release it.

**Comment by mauritsvanrees** [comment]
I have released 1.16.3.

**Comment by vincentfretin** [comment]
I still maintain old Plone for some customers.
FYI This is not an issue on Plone 4.3.x (I looked for Plone 4.3.4.1 and 4.3.10), we have a ceiling at 2500, there is no `Products/Archetypes/browser/datecomponents.py` file, but we have `Products/CMFPlone/skins/plone_scripts/date_components_support.py` instead.

For Plone 4.1.6
`Products.CMFPlone-4.1.6-py2.7.egg/Products/CMFPlone/skins/plone_scripts/date_components_support.py`
and
`plone.app.form-2.0.5-py2.7.egg/plone/app/form/widgets/datecomponents.py`
includes
```
PLONE_CEILING=DateTime(2021,0) # 2020-12-31
```
so be sure to check that and modify the files with
```
PLONE_CEILING = DateTime(2500, 0) # 2499-12-31
```
if you maintain old Plone versions for some customers.

**Comment by vincentfretin** [comment]
on Plone 4.1.6 `@@date_components_support` defined in `plone.app.form-2.0.5-py2.7.egg/plone/app/form/widgets/datecomponents.py` seems to only be used by `Products/CMFPlone/skins/plone_templates/calendar_macros.pt` that is used for the Archetypes datetime widget.
`Products/CMFPlone/skins/plone_scripts/date_components_support.py` doesn't seem to be used.
I verified myself the issue by using `PLONE_CEILING=DateTime(2019,0)` in `plone/app/form/widgets/datecomponents.py`, it's indeed a bad bug when creating a new Archetypes content or editing an existing one. :D

**Comment by vincentfretin** [comment]
I monkey patched it from my policy package for this customer like this:
```
from DateTime.DateTime import DateTime                                          
import plone.app.form.widgets.datecomponents                                    
plone.app.form.widgets.datecomponents.PLONE_CEILING = DateTime(2500, 0) # 2499-12-31
```

done. :) This Plone install will survive some more years. :D

**Comment by mauritsvanrees** [comment]
I have already released `plone.app.form` 1.1.10 with a [fix from @iham](https://github.com/plone/plone.app.form/issues/17).
If someone makes PRs for more packages and branches, or perhaps in this simple case a direct fix on a branch, I am okay with doing more package releases.
Well, not for `Products.CMFPlone`, but others should be fine.

**Comment by lucamancini** [comment]
> I still maintain old Plone for some customers.
> FYI This is not an issue on Plone 4.3.x (I looked for Plone 4.3.4.1 and 4.3.10), we have a ceiling at 2500, there is no `Products/Archetypes/browser/datecomponents.py` file, but we have `Products/CMFPlone/skins/plone_scripts/date_components_support.py` instead.
> 
> For Plone 4.1.6
> `Products.CMFPlone-4.1.6-py2.7.egg/Products/CMFPlone/skins/plone_scripts/date_components_support.py`
> and
> `plone.app.form-2.0.5-py2.7.egg/plone/app/form/widgets/datecomponents.py`
> includes
> 
> ```
> PLONE_CEILING=DateTime(2021,0) # 2020-12-31
> ```
> 
> so be sure to check that and modify the files with
> 
> ```
> PLONE_CEILING = DateTime(2500, 0) # 2499-12-31
> ```
> 
> if you maintain old Plone versions for some customers.

For Plone 4.2?
I did the same in 
products.cmfplone-4.2.0.1-py2.6.egg\Products\CMFPlone\skins\plone_scripts/date_components_support.py`
and
plone.app.form-2.1-py2.6.egg\plone\app\form\widgets/datecomponents.py

**Comment by lucamancini** [comment]
And it works!
Thank you

**Comment by mauritsvanrees** [comment]
For `Products.Archetypes` I have released fixes: 1.10.20 (Plone 5.0), 1.15.9 (Plone 5.1), and 1.16.4 (Plone 5.2).

**Comment by barbiebrega** [comment]
I have altered the plone_ceiling and it works.
BUT I recently discovered that news published in 2021 are not being indexed by google.
Anyone knows something about that?
